packages/headless/src/controllers/instant-results/instant-results.ts

```diff
@@ -0,0 +1,131 @@
+import {SearchEngine} from '../../app/search-engine/search-engine';
+import {Result} from '../../api/search/search/result';
+import {InstantResultSection} from '../../state/state-sections';
+import {buildController, Controller} from '../controller/headless-controller';
+import {instantResults} from '../../app/reducers';
+import {loadReducerError} from '../../utils/errors';
+import {validateOptions} from '../../utils/validate-payload';
+import {
+  fetchInstantResults,
+  registerInstantResults,
+  updateInstantResultsQuery,
+} from '../../features/instant-results/instant-results-actions';
+import {
+  InstantResultOptions,
+  instantResultsOptionsSchema,
+} from './instant-results-options';
+import {SearchAPIErrorWithStatusCode} from '../../api/search/search-api-error-response';
+import {SerializedError} from '@reduxjs/toolkit';
+
+export interface InstantResultProps {
+  options: InstantResultOptions;
+}
+
+export interface InstantResults extends Controller {
+  /**
+   * Updates the instant results query and shows instant results for it.
+   *
+   * @param q - The string value to get instant results for.
```

Comment:
```suggestion
   * @param q - The query to get instant results for. For more precise instant results, query suggestions are recommended.
```

@olamothe I've added a sentence here based on what you said above as it made total sense to me 🙏 The only thing is a bit unclear at the moment is you said "in order to get good instant results"; I understood it as "more precise instant results", but this may be wrong. Feel free to correct me and/or comment on this subject 😇